pkg/azure/testhelp/fakes/machineresources.go

```diff
@@ -151,6 +151,14 @@ func (m *MachineResources) UpdateDataDisksDeleteOpt(deleteOpt *armcompute.DiskDe
 	}
 }
 
+// AttachDataDisk attaches a data disk to the VM
+func (m *MachineResources) AttachDataDisk(spec api.AzureProviderSpec, diskName string, lun int32, deleteOption armcompute.DiskDeleteOptionTypes) {
+	dataDisk := createDataDisk(lun, "None", &deleteOption, 20, testhelp.StorageAccountType, diskName)
+	d := createDiskResource(spec, diskName, m.VM.ID, nil)
+	m.DataDisks[diskName] = d
```

Comment:
What if there is already a disk with the same name? Since this is test helper code its fine to do this but just to make it safer i would say error out if there is already something so that the test writer will be aware of such problems and does not have to spend time in diagnosing the issue.